### ISSUE TITLE
sparse_categorical_crossentropy: better documentation and general dimension handling 

### ISSUE THREAD

**Issue by DSLituiev** [keras-team/keras#8336]
System info:
keras: 2.0.8
tensorflow: 1.5.0-dev20171031

The documentation to keras `sparse_categorical_crossentropy` seems to be quite obscure when compared to one of [`tensorflow.nn.sparse_softmax_cross_entropy_with_logits`](https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits). It is not clear what dimensionality of tensors are expected. 

It breaks on a simple segmentation problem with logits of dimension `[bs, height,width, channels]` and target of `[bs, height,width, 1]`, see [gist](https://gist.github.com/DSLituiev/740813f65c785c13ae294ec5d46bfb17)

There is a corner case handling for tensors of dimensionality 3 [in line 2895](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2895), but what about images? I have a solution to my case, but without a clear specification of dimensionality or test case, I am not sure whether it will work with the dimension of 3.

**Comment by MADHAVAN001** [comment]
It would be beneficial to provide the intuition behind how this is working for building better models.

The doc can be a little more verbose.
